dbms/src/Functions/FunctionsRegexp.h

```diff
@@ -973,15 +1136,383 @@ class FunctionStringRegexp : public FunctionStringRegexpBase
+                        continue;
+                    }
+                    null_map[i] = 0;
+                    expr_param.getStringRef(i, expr_ref);
+                    pat = pat_param.getString(i);
+                    if (unlikely(pat.empty()))
+                        throw Exception(EMPTY_PAT_ERR_MSG);
+                    pat = fmt::format("({})", pat);
+                    GET_POS_VALUE(i)
+                    GET_OCCUR_VALUE(i)
+                    GET_RET_OP_VALUE(i)
+                    match_type = match_type_param.getString(i);
+                    auto regexp = createRegexpWithMatchType(pat, match_type, collator);
+                    vec_res[i] = regexp.instr(expr_ref.data, expr_ref.size, pos, occur, ret_op);
+                }
+
+                res_arg.column = ColumnNullable::create(std::move(col_res), std::move(nullmap_col));
+            }
+            else
+            {
+                // Process pure vector columns without memorized regexp
+                for (size_t i = 0; i < col_size; ++i)
+                {
+                    expr_param.getStringRef(i, expr_ref);
+                    pat = pat_param.getString(i);
+                    if (unlikely(pat.empty()))
+                        throw Exception(EMPTY_PAT_ERR_MSG);
+                    pat = fmt::format("({})", pat);
+                    GET_POS_VALUE(i)
+                    GET_OCCUR_VALUE(i)
+                    GET_RET_OP_VALUE(i)
+                    match_type = match_type_param.getString(i);
+                    auto regexp = createRegexpWithMatchType(pat, match_type, collator);
+                    vec_res[i] = regexp.instr(expr_ref.data, expr_ref.size, pos, occur, ret_op);
+                }
+
+                res_arg.column = std::move(col_res);
+            }
+        }
+
+#undef GET_RET_OP_VALUE
+#undef GET_OCCUR_VALUE
+#undef GET_POS_VALUE
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        // Do something related with nullable columns
+        NullPresence null_presence = getNullPresense(block, arguments);
+
+        if (null_presence.has_null_constant)
+        {
+            block.getByPosition(result).column = block.getByPosition(result).type->createColumnConst(block.rows(), Null());
+            return;
+        }
+
+        const ColumnPtr & col_expr = block.getByPosition(arguments[0]).column;
+        const ColumnPtr & col_pat = block.getByPosition(arguments[1]).column;
+
+        size_t arg_num = arguments.size();
+        auto & RES_ARG_VAR_NAME = block.getByPosition(result);
+
+        ColumnPtr col_pos;
+        ColumnPtr col_occur;
+        ColumnPtr col_return_option;
+        ColumnPtr col_match_type;
+
+        // Go through cases to get arguments
+        switch (arg_num)
+        {
+        case REGEXP_INSTR_MAX_PARAM_NUM:
+            col_match_type = block.getByPosition(arguments[5]).column;
+        case REGEXP_MIN_PARAM_NUM + 3:
```

Comment:
I think it's more understandable by using `case REGEXP_INSTR_MAX_PARAM_NUM - XXX` instead of `case REGEXP_MIN_PARAM_NUM + XXX`

Reply:
> I think it's more understandable by using `case REGEXP_INSTR_MAX_PARAM_NUM - XXX` instead of `case REGEXP_MIN_PARAM_NUM + XXX`

okk

Reply:
> I think it's more understandable by using `case REGEXP_INSTR_MAX_PARAM_NUM - XXX` instead of `case REGEXP_MIN_PARAM_NUM + XXX`

Done